### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by hsquek**
Added tests for tutor model and tutor authentication (signup and login)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "main": "index.js",
   "scripts": {
     "start": "node index.js",
-    "dev-start": "nodemon index.js"
+    "dev-start": "nodemon index.js",
+    "test": "SIGNUP_PASSWORD=testingpassphrase NODE_ENV=test ./node_modules/mocha/bin/mocha"
```
Comment (melvinthemok) [inline package.json]: is a `SIGNUP_PASSWORD` different from what is defined in your `.env` file necessary?

File: package.json [hunk 2]
```diff
@@ -37,5 +38,11 @@
     "nodemon": "^1.11.0",
     "passport": "^0.3.2",
     "passport-local": "^1.0.0"
+  },
+  "devDependencies": {
+    "chai": "^4.1.2",
+    "chai-as-promised": "^7.1.1",
+    "mocha": "^4.0.0",
+    "supertest": "^3.0.0"
```
Comment (melvinthemok) [inline package.json]: if you figure out how to deploy to heroku with devDependencies, please move `ejs-lint`, `morgan`, and `nodemon` here too!

File: test/tutor_tests.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
+var expect = require('chai').expect
+var should = require('chai').should()
+var chai = require('chai')
+var chaiAsPromised = require('chai-as-promised')
```
Comment (melvinthemok) [inline test/tutor_tests.js]: are there a lot of promises to be tested? trying to understand rationale for including this

File: test/tutor_tests.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+      var tutor = new Tutor(createTutorWithFaultyAttr(faultyFields))
+      tutor.save().should.eventually.be.rejectedWith(Error).notify(done)
+    })
+
+    it('Should successfully save tutor if the appropriate params are passed', function (done) {
+      var tutor = new Tutor(createTutorWithFaultyAttr({}))
+      tutor.save().should.eventually.be.fulfilled.notify(done)
+    })
+
+    it('Should hash the given password', function (done) {
```
Comment (melvinthemok) [inline test/tutor_tests.js]: this test is failing for me

File: package.json (hunk 1 above)
Comment (hsquek) [reply package.json]: I didn't want you guys to have to change the `SIGNUP_PASSWORD` in order to run the tests, but yeah I get what you mean.

Made a slight tweak, no longer necessary.

File: test/tutor_tests.js (hunk 3 above)
Comment (hsquek) [reply test/tutor_tests.js]: Thought this improved readability, but will remove for consistency.

**Comment by melvinthemok** [comment]
looks good to me! please go ahead to merge and deploy - with deployment in general, it would be good to check that everything is working fine in production after deploying

File: test/tutor_tests.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    })
+
+    after(function (done) {
+      mongoose.connection.collections.tutors.drop(function () {
+        done()
+      })
+    })
+
+    it('Should redirect to sign up page if an invalid field is given', function (done) {
+      var signUpPassword = process.env.SIGNUP_PASSWORD
```
Comment (melvinthemok) [inline test/tutor_tests.js]: @hs just noticed that the tests still pass even if `process.env.SIGNUP_PASSWORD` is `undefined`

Comment (hsquek) [reply test/tutor_tests.js]: @melvinthemok i'll add a hook to test for the presence of process.env.SIGNUP_PASSWORD in the next PR for the next set of tests

**Pull request merged**